### PR TITLE
Replace Bits with uint128_t in phase 2

### PR DESCRIPTION
**Pull request by rostislav**
Improve CPU performance of phase 2 by 10% when using bitfield.

### CODE REVIEW

**Review by hoffmang9 — APPROVED** [review]
+1 - will merge once the k32 confirms same speed or better.

File: src/phase2.hpp [hunk 1]
```diff
@@ -163,61 +167,49 @@ Phase2Results RunPhase2(
...
             std::tie(entry_pos, entry_offset) = index.lookup(entry_pos, entry_offset);
+            entry_pos_offset = (entry_pos << kOffsetSize) | entry_offset;
 
+            uint8_t bytes[16];
             if (table_index == 7) {
                 // table 7 is already sorted by pos, so we just rewrite the
                 // pos and offset in-place
+                uint128_t new_entry = (uint128_t)entry_f7 << f7_shift;
+                new_entry |= (uint128_t)entry_pos_offset << t7_pos_offset_shift;
+                Util::IntTo16Bytes(bytes, new_entry);
```
Comment (arvidn) [inline src/phase2.hpp:200]: is there a good reason to not do this byteswap in-place?
Also, if we always write a full 16 byte entries here, is there a good reason not to leave it in the host-endian?

Comment (rostislav) [reply src/phase2.hpp:200]: We don't always write 16 bytes here, the entry size is determined by `EntrySizes::GetMaxEntrySize`, which returns `3*k+1` bits rounded up to whole bytes. There's also a convention in the codebase that all integers are encoded as big-endian before being written to disk, including in `SortManager` which assumes this as well for sorting entries.

**Comment by hoffmang9** [comment]
Early results are that this is 26% faster in phase 2 in bitfield. Merging and will then pull into chia-blockchain once we have final numbers.

Nicely done @rostislav

**Comment by mariano54** [comment]
Nice. Does this put bitfield above non-bitfield, with 4GB of ram?

I would love to see a run of 4GB ram bitfield to compare. So then we can get finally rid of the old code.

**Comment by hoffmang9** [comment]
@justinengland will be able to do a non bitfield run on same hardware during the day tomorrow to get a bottom line comparison between new bitfield and classic no bitfield.

**Pull request merged**